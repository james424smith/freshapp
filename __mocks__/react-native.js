import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
