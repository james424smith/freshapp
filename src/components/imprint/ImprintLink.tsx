import React from "react";
import { View } from "react-native";
import Text from "../StyledText";
import styles from "../../styles/AuthStyles";
import { TouchableOpacity } from "react-native-gesture-handler";

const IMPRINT_SCREEN = "Imprint";

type Props = { navigate: (arg: string) => void };

const ImprintLink = (props: Props) => {
  const { navigate } = props;
  return (
    <View style={styles.imprintLinkContainer}>
      <View style={styles.imprintLinkView}>
        <TouchableOpacity
          testID="imprint-button"
          onPress={() => navigate(IMPRINT_SCREEN)}
        >
          <Text style={styles.imprintLink}>Imprint</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default ImprintLink;
