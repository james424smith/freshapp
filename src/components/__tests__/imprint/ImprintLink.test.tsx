import React from "react";
import ImprintLink from "../../imprint/ImprintLink";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

describe("ImprintLink snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui when imprint", () => {
    const navigation = jest.fn();

    const { toJSON, getByTestId } = render(
      <ImprintLink navigate={navigation} />
    );
    fireEvent.press(getByTestId("imprint-button"));
    expect(navigation).toHaveBeenCalledWith("Imprint");
    expect(toJSON()).toMatchSnapshot();
  });
});
