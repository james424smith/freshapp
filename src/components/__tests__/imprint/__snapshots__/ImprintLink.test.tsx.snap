// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImprintLink snapshot test should render correct ui when imprint 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      testID="imprint-button"
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Object {
                "color": "#78849E",
                "textDecorationLine": "underline",
              },
            ]
          }
        >
          Imprint
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
